src/main/resources/static/assets/js/sdh.js

```diff
@@ -73,8 +75,84 @@
 		    _isShowDetails(field) {
 		    	return field.key === 'show_details';
 		    },
+		    _isEdit(field) {
+		    	return field.key === 'edit';
+		    },
 		    _formatDate(dateString) {
 		    	return dateString.replace(/ \d\d:\d\d:\d\d\.\d/, '');
+		    },
+		    edit(item, index, target) {
+		    	this.modalEditRow.domain = this.domainOptions.filter(it => it.description === item.domain)[0].id;
+		    	if (item.id === "") {
+		    		this.modalEditRow.title = 'New Clinical Review';
+		    	} else {
+		    		this.modalEditRow.title = 'Edit Clinical Review';
+		    		this.modalEditRow.id = item.id;
+		    	      this.modalEditRow.prioritySelected = this.modalEditRow.priorityOptions.filter(it => it.text === item.clinician_priority)[0].value;
+			        this.modalEditRow.readinessSelected = this.modalEditRow.readinessOptions.filter(it => it.text === item.patient_readiness)[0].value;
+		    	}
+				this.$root.$emit('bv::show::modal', 'modalEditRow', target);
+		    },
+		    resetModal() {
+		        this.modalEditRow.title = '';
+		        this.modalEditRow.id = '';
+		        this.modalEditRow.domain = '';
+		        this.modalEditRow.prioritySelected = '';
+		        this.modalEditRow.readinessSelected = '';
```

Comment:
There's a fair amount of repetition around manipulating the `modalEditRow` object. It seems like things could be simplified by extracting a class. Then, for instance, `resetModal` would boil down to `this.modalEditRow = new ModalEditRow()`. In other cases, logic that you have inline could be encapsulated in methods.